pkg/apis/serving/v1beta1/inference_service_validation.go

```diff
@@ -80,6 +85,18 @@ func (isvc *InferenceService) ValidateCreate() error {
 	return nil
 }
 
+// Validate scaling options component extensions
+func validateAutoScalingCompExtension(annotations map[string]string, compExtSpec *ComponentExtensionSpec) error {
+	deploymentMode := annotations["serving.kserve.io/deploymentMode"]
```

Comment:
we also need to get the deploymentMode from configmap if  `deploymentMode` is set to `RawDeployment` so this annotation may not present in that case.

Reply:
Yes, will check...

Reply:
Seems there's a helper function for this already you can use

`isvcutils.GetDeploymentMode(annotations, deployConfig)`

https://github.com/kserve/kserve/blob/master/pkg/controller/v1beta1/inferenceservice/utils/utils.go